File: python/high-scores/high_scores.py
"""
Functions to retrieve score informations
"""
def latest(scores):
    """
    get the latest score
    """
    return scores[-1]


def personal_best(scores):
    """
    get the highest score
    """
    return max(scores)


def personal_top_three(scores):
    """
    get the top three highest score
    """
    return sorted(scores, reverse=True)[:3]

File: ruby/high-scores/high_scores.rb
class HighScores
  attr_reader :scores
  def initialize(scores)
    @scores = scores
  end

  def latest
    @scores.last
  end

  def personal_best
    @scores.max
  end

  def personal_top_three
    @scores.sort.reverse[0..2]
  end

  def latest_is_personal_best?
    @scores.last == personal_best
  end
end